crates/ruma-signatures/src/functions.rs

```diff
@@ -48,7 +48,10 @@ fn allowed_content_keys_for(event_type: &str, version: &RoomVersionId) -> &'stat
             _ => &["membership"],
         },
         "m.room.create" => &["creator"],
-        "m.room.join_rules" => &["join_rule"],
+        "m.room.join_rules" => match version {
+            RoomVersionId::V8 | RoomVersionId::V9 => &["join_rule", "allow"],
```

Comment:
We should not match the version here and instead use the constants defined in room versions cc @DevinR528 

Maybe this can be done after this MR has been merged

Reply:
I think that sounds reasonable, although I think it would create a circular dep so the room version stuff would have to be moved to `ruma-common` maybe? Since this is outside of state-res @jplatte  would probably have to weigh in anyways.